python-package/setup.py

```diff
@@ -28,6 +28,7 @@
       install_requires=[
           'numpy',
           'scipy',
+          'scikit-learn',
```

Comment:
we have `sklearn.py` in the `python-package/xgboost`, and import from '__init__.py' https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/__init__.py#L15 ,  if users have no `scikit-learn`, `sklearn.py` would raise an error for catch. There are couple of issues about this dependency, since people have asked, I put it to the dependency. Do we want to remove it?